flocker/control/httpapi.py

```diff
@@ -812,11 +818,15 @@ def replace_configuration(self, applications, deployment):
         :param deployment: Configuration of which applications run on
             which nodes.
         """
+        # XXX we're going to have to deal with UUIDs better,
+        # somehow... look up state of cluster? Change deployment
+        # configuration format?
         try:
             configuration = FigConfiguration(applications)
             if not configuration.is_valid_format():
                 configuration = FlockerConfiguration(applications)
             return self.persistence_service.save(model_from_configuration(
+                self.cluster_state_service.as_deployment(),
```

Comment:
So the `Deployment` instance is returned and contains `Node`'s whose UUIDs are generated from the supplied IP addresses.
That `Deployment` replaces the existing configuration and the connected agents are notified of a configuration change.

Next the convergence agents on each node need to know the UUID of the node that they're running on so that they can identify which parts of the configuration concern them...and the agents get their UUID from the flocker configuration file whose location is defined in the systemd service file...